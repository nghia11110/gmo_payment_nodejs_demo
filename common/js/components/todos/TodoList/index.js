import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';
import { TodoItem } from '@components/todos';
import classnames from 'classnames';
import css from './index.scss';

const TodoList = props => {
  const { className, onChange, onRemove, todos: { todos } } = props;

  return (
    <List divided className={classnames(css.todos, className)}>
      {todos.map((todo, idx) => (
        <TodoItem
          key={idx}
          todo={todo}
          onRemove={onRemove}
          onChange={onChange}
        />
      ))}
    </List>
  );
};

TodoList.propTypes = {
  todos: PropTypes.object.isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func,
  onRemove: PropTypes.func
};

export default TodoList;
