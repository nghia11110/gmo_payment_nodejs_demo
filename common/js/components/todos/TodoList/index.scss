.todos {
  width: 300px;
  margin: 0;
  padding: 0;
  list-style: none;
}
