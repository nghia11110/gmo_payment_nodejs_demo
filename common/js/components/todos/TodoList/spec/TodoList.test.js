import React from 'react';
import TodoList from '../index';
import { shallow } from 'enzyme';

describe('TodoList', () => {
  let todos = {
    isFetched: true,
    todos: [
      {
        id: 1,
        text: 'Learn React',
        completed: true
      },
      {
        id: 2,
        text: 'Learn Redux',
        completed: true
      }
    ]
  };


  it('renders correctly', () => {
    const component = shallow(<TodoList todos={todos} />);
    expect(component).toMatchSnapshot();
  });
});
