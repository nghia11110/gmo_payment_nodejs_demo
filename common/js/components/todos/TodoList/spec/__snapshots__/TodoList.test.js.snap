// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders correctly 1`] = `
<List
  className="todos"
  divided={true}
>
  <TodoItem
    key="0"
    onChange={[Function]}
    onRemove={[Function]}
    todo={
      Object {
        "completed": true,
        "id": 1,
        "text": "Learn React",
      }
    }
  />
  <TodoItem
    key="1"
    onChange={[Function]}
    onRemove={[Function]}
    todo={
      Object {
        "completed": true,
        "id": 2,
        "text": "Learn Redux",
      }
    }
  />
</List>
`;
